### PR TITLE
[plugin] Support all web clients

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

**Comment by grqz** [comment]
maybe do it on #19 
or do you want to merge it now?

**Comment by Brainicism** [comment]
We can just wait until after #19 is merged since its not really related to the refactor, and itll be easier to change after

**Review by grqz — APPROVED** [review]

**Pull request merged**
